### PR TITLE
[settings] allow grouping of settings to allow removing at once

### PR DESCRIPTION
**Pull request by 3nids**

### CODE REVIEW

**Comment by 3nids** [comment]
This is ready for review.
I have tested adding/deleting connection with this PR. The connections are successfully read from an older QGIS.

File: python/gui/auto_generated/qgsnewhttpconnection.sip.in [hunk 1]
```diff
@@ -43,7 +43,7 @@ information for an HTTP Server for WMS, etc.
 
     QgsNewHttpConnection( QWidget *parent /TransferThis/ = 0,
                           QgsNewHttpConnection::ConnectionTypes types = ConnectionWms,
-                          const QString &baseKey = "qgis/connections-wms/",
+                          const QString &serviceName = "WMS",
```
Comment (nyalldawson) [inline python/gui/auto_generated/qgsnewhttpconnection.sip.in:46]: This is an api break, isn't it?

File: tests/src/python/test_qgsgeonodeconnection.py [hunk 2]
```diff
@@ -35,6 +35,8 @@ def setUpClass(cls):
         settings = QgsSettings()
         key = QgsGeoNodeConnectionUtils.pathGeoNodeConnection() + '/test/'
 
+        print("xxx" + key)
```
Comment (nyalldawson) [inline tests/src/python/test_qgsgeonodeconnection.py:38]: remove

File: src/providers/wfs/qgswfsconstants.h [hunk 3]
```diff
@@ -54,12 +54,12 @@ struct QgsWFSConstants
...
-  static const QString SETTINGS_MAXNUMFEATURES;
-  static const QString SETTINGS_PAGING_ENABLED;
-  static const QString SETTINGS_PAGE_SIZE;
-  static const QString SETTINGS_WFST_1_1_PREFER_COORDINATES;
+//  static const QString CONNECTIONS_WFS;
+//  static const QString SETTINGS_VERSION;
+//  static const QString SETTINGS_MAXNUMFEATURES;
+//  static const QString SETTINGS_PAGING_ENABLED;
+//  static const QString SETTINGS_PAGE_SIZE;
+//  static const QString SETTINGS_WFST_1_1_PREFER_COORDINATES;
```
Comment (nyalldawson) [inline src/providers/wfs/qgswfsconstants.h:62]: remove

File: src/providers/wfs/qgswfsconstants.cpp [hunk 4]
```diff
@@ -44,9 +44,9 @@ const QString QgsWFSConstants::URI_PARAM_WFST_1_1_PREFER_COORDINATES( "preferCoo
...
-const QString QgsWFSConstants::SETTINGS_MAXNUMFEATURES( QStringLiteral( "maxnumfeatures" ) );
-const QString QgsWFSConstants::SETTINGS_PAGING_ENABLED( QStringLiteral( "pagingenabled" ) );
-const QString QgsWFSConstants::SETTINGS_PAGE_SIZE( QStringLiteral( "pagesize" ) );
-const QString QgsWFSConstants::SETTINGS_WFST_1_1_PREFER_COORDINATES( QStringLiteral( "preferCoordinatesForWfsT11" ) );
+//const QString QgsWFSConstants::CONNECTIONS_WFS( QStringLiteral( "qgis/connections-wfs/" ) );
+//const QString QgsWFSConstants::SETTINGS_VERSION( QStringLiteral( "version" ) );
+//const QString QgsWFSConstants::SETTINGS_MAXNUMFEATURES( QStringLiteral( "maxnumfeatures" ) );
+//const QString QgsWFSConstants::SETTINGS_PAGING_ENABLED( QStringLiteral( "pagingenabled" ) );
+//const QString QgsWFSConstants::SETTINGS_PAGE_SIZE( QStringLiteral( "pagesize" ) );
+//const QString QgsWFSConstants::SETTINGS_WFST_1_1_PREFER_COORDINATES( QStringLiteral( "preferCoordinatesForWfsT11" ) );
```
Comment (nyalldawson) [inline src/providers/wfs/qgswfsconstants.cpp:52]: remove

File: src/core/qgsowsconnection.h [hunk 5]
```diff
@@ -38,6 +40,44 @@ class CORE_EXPORT QgsOwsConnection : public QObject
...
+    static const inline QgsSettingsEntryBool settingsConnectionIgnoreGetFeatureInfoURI = QgsSettingsEntryBool( QStringLiteral( "connections-%1/%2/ignoreGetFeatureInfoURI" ), QgsSettings::Prefix::QGIS, false ) SIP_SKIP;
+    static const inline QgsSettingsEntryBool settingsConnectionSmoothPixmapTransform = QgsSettingsEntryBool( QStringLiteral( "connections-%1/%2/smoothPixmapTransform" ), QgsSettings::Prefix::QGIS, false ) SIP_SKIP;
+    static const inline QgsSettingsEntryBool settingsConnectionReportedLayerExtents = QgsSettingsEntryBool( QStringLiteral( "connections-%1/%2/reportedLayerExtents" ), QgsSettings::Prefix::QGIS, false ) SIP_SKIP;
+    static const inline QgsSettingsEntryEnumFlag<DpiMode> settingsConnectionDpiMode = QgsSettingsEntryEnumFlag<DpiMode>( QStringLiteral( "connections-%1/%2/dpiMode" ), QgsSettings::Prefix::QGIS, DpiMode::All, QString(), Qgis::SettingsOption::SaveEnumFlagAsInt ) SIP_SKIP;
+    static const inline QgsSettingsEntryString settingsConnectionMaxnumFeatures = QgsSettingsEntryString( QStringLiteral( "connections-%1/%2/maxnumfeatures" ), QgsSettings::Prefix::QGIS ) SIP_SKIP;
```
Comment (nyalldawson) [inline src/core/qgsowsconnection.h:67]:
```suggestion
    static const inline QgsSettingsEntryString settingsConnectionMaxNumFeatures = QgsSettingsEntryString( QStringLiteral( "connections-%1/%2/maxnumfeatures" ), QgsSettings::Prefix::QGIS ) SIP_SKIP;
```

File: src/core/qgsowsconnection.h [hunk 6]
```diff
@@ -38,6 +40,44 @@ class CORE_EXPORT QgsOwsConnection : public QObject
...
+    static const inline QgsSettingsEntryString settingsConnectionPagesize = QgsSettingsEntryString( QStringLiteral( "connections-%1/%2/pagesize" ), QgsSettings::Prefix::QGIS ) SIP_SKIP;
+    static const inline QgsSettingsEntryBool settingsConnectionPagingenabled = QgsSettingsEntryBool( QStringLiteral( "connections-%1/%2/pagingenabled" ), QgsSettings::Prefix::QGIS, true ) SIP_SKIP;
```
Comment (nyalldawson) [inline src/core/qgsowsconnection.h:69]:
```suggestion
    static const inline QgsSettingsEntryBool settingsConnectionPagingEnabled = QgsSettingsEntryBool( QStringLiteral( "connections-%1/%2/pagingenabled" ), QgsSettings::Prefix::QGIS, true ) SIP_SKIP;
```

File: src/core/qgsowsconnection.h [hunk 7]
```diff
@@ -38,6 +40,44 @@ class CORE_EXPORT QgsOwsConnection : public QObject
 
   public:
 
+    /**
+     * DpiMode enum
+     * \since QGIS 3.26
+     */
+    enum class DpiMode
```
Comment (nyalldawson) [inline src/core/qgsowsconnection.h:47]: Would this be better in Qgis?

File: python/gui/auto_generated/qgsnewhttpconnection.sip.in (hunk 1 above)
Comment (3nids) [reply python/gui/auto_generated/qgsnewhttpconnection.sip.in:46]: no, this is handled here https://github.com/qgis/QGIS/pull/48192/files?w=1#diff-72f4e9193431ef42d291a483c045825bfacb9df7e5691fd065e091ec47d636b8R40

File: src/core/qgsowsconnection.h (hunk 7 above)
Comment (3nids) [reply src/core/qgsowsconnection.h:47]: moved

File: python/gui/auto_generated/qgsnewhttpconnection.sip.in (hunk 1 above)
Comment (3nids) [reply python/gui/auto_generated/qgsnewhttpconnection.sip.in:46]: image

Comment (nyalldawson) [reply python/gui/auto_generated/qgsnewhttpconnection.sip.in:46]: I mean the rename of the keyword -- if some code is using `QgsNewHttpConnection(..., baseKey='xxxxx')` then it'll now raise an exception.

File: src/core/qgis.h [hunk 8]
```diff
@@ -1386,6 +1386,20 @@ class CORE_EXPORT Qgis
...
+    /**
+     * DpiMode enum
+     * \since QGIS 3.26
+     */
+    enum class DpiMode
+    {
+      All = 7, //!< All
+      Off = 0, //!< Off
+      QGIS = 1, //!< QGIS
+      UMN = 2, //!< UMN
```
Comment (nyalldawson) [inline src/core/qgis.h:1398]: Just curious -- what actually is "UMN"? :laughing:

File: python/gui/auto_generated/qgsnewhttpconnection.sip.in (hunk 1 above)
Comment (3nids) [reply python/gui/auto_generated/qgsnewhttpconnection.sip.in]: you're right. I've renamed the arg for python

File: src/core/qgis.h (hunk 8 above)
Comment (3nids) [reply src/core/qgis.h:1398]: I have no idea!

**Review by nyalldawson — APPROVED** [review]

File: src/core/qgis.h (hunk 8 above)
Comment (landryb) [reply src/core/qgis.h:1398]: Probably "University Of Minnesota", where mapserver comes from :)

**Pull request merged**